scrapy/log.py

```diff
@@ -1,161 +1,10 @@
-        # ignore spiders' events in observers without crawler
-        return
-
-    lvlname = level_names.get(level, 'NOLEVEL')
-    message = ev.get('message')
-    if message:
-        message = [unicode_to_str(x, encoding) for x in message]
-        if prepend_level:
-            message[0] = "%s: %s" % (lvlname, message[0])
-        ev['message'] = message
-
-    why = ev.get('why')
-    if why:
-        why = unicode_to_str(why, encoding)
-        if prepend_level:
-            why = "%s: %s" % (lvlname, why)
-        ev['why'] = why
-
-    fmt = ev.get('format')
-    if fmt:
-        fmt = unicode_to_str(fmt, encoding)
-        if prepend_level:
-            fmt = "%s: %s" % (lvlname, fmt)
-        ev['format'] = fmt
-
-    return ev
-
-def _get_log_level(level_name_or_id):
-    if isinstance(level_name_or_id, int):
-        return level_name_or_id
-    elif isinstance(level_name_or_id, basestring):
-        return globals()[level_name_or_id]
-    else:
-        raise ValueError("Unknown log level: %r" % level_name_or_id)
-
-def start(logfile=None, loglevel='INFO', logstdout=True, logencoding='utf-8', crawler=None):
-    loglevel = _get_log_level(loglevel)
-    file = open(logfile, 'a') if logfile else sys.stderr
-    log_observer = ScrapyFileLogObserver(file, loglevel, logencoding, crawler)
-    _oldshowwarning = warnings.showwarning
-    log.startLoggingWithObserver(log_observer.emit, setStdout=logstdout)
-    # restore warnings, wrongly silenced by Twisted
-    warnings.showwarning = _oldshowwarning
-    return log_observer
-
-def msg(message=None, _level=INFO, **kw):
-    kw['logLevel'] = kw.pop('level', _level)
-    kw.setdefault('system', 'scrapy')
-    if message is None:
-        log.msg(**kw)
-    else:
-        log.msg(message, **kw)
-
-def err(_stuff=None, _why=None, **kw):
-    kw['logLevel'] = kw.pop('level', ERROR)
-    kw.setdefault('system', 'scrapy')
-    log.err(_stuff, _why, **kw)
-
-def start_from_settings(settings, crawler=None):
-    if settings.getbool('LOG_ENABLED'):
-        return start(settings['LOG_FILE'], settings['LOG_LEVEL'], settings['LOG_STDOUT'],
-            settings['LOG_ENCODING'], crawler)
-
-def scrapy_info(settings):
-    msg("Scrapy %s started (bot: %s)" % (scrapy.__version__,
-        settings['BOT_NAME']))
-
-    msg("Optional features available: %s" % ", ".join(scrapy.optional_features),
-        level=INFO)
-
-    d = dict(overridden_settings(settings))
-    msg(format="Overridden settings: %(settings)r", settings=d, level=INFO)
-
-def start_from_crawler(crawler):
-    return start_from_settings(crawler.settings, crawler)
+import warnings
+from scrapy.exceptions import ScrapyDeprecationWarning
+warnings.warn("Module `scrapy.log` has been deprecated, Scrapy relies now on "
+              "the builtin Python library for logging. Read the updated "
+              "logging entry in the documentation to learn more.",
+              ScrapyDeprecationWarning, stacklevel=2)
```

Comment:
I think it worths it to add log.msg and log.err aliases for backwards compatibility.